pcsx2/GS.cpp

```diff
@@ -138,13 +139,24 @@ __fi void gsWrite8(u32 mem, u8 value)
 	GIF_LOG("GS write 8 at %8.8lx with data %8.8lx", mem, value);
 }
 
+static void GetColorBurst(u32 mem, u32 value)
+{
+	if (mem != GS_SMODE1)
+		return;
+
+	GSRegSMODE1 Register;
+	Register.SMODE1 = value;
+	if(u32 colorburst = Register.CMOD)
+		gsSetVideoMode(colorburst == 3 ? GS_VideoMode::PAL : GS_VideoMode::NTSC);
+}
+
 //////////////////////////////////////////////////////////////////////////
 // GS Write 16 bit
 
 __fi void gsWrite16(u32 mem, u16 value)
 {
 	GIF_LOG("GS write 16 at %8.8lx with data %8.8lx", mem, value);
-
+	GetColorBurst(mem, value);
```

Comment:
First you won't need to add the mem parameter. Then you understand that GetColorBurst is only for handling a single register and not a generic function that could impact everything.